centreon/src/Core/ResourceAccess/Application/UseCase/DeleteRule/DeleteRule.php

```diff
@@ -0,0 +1,148 @@
+use Core\ResourceAccess\Application\Repository\ReadResourceAccessRepositoryInterface;
+use Core\ResourceAccess\Application\Repository\WriteResourceAccessRepositoryInterface;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+use Core\Security\AccessGroup\Domain\Model\AccessGroup;
+
+final class DeleteRule
+{
+    use LoggerTrait;
+    public const AUTHORIZED_ACL_GROUPS = ['customer_admin_acl'];
+
+    /**
+     * @param ReadResourceAccessRepositoryInterface $readRepository
+     * @param WriteResourceAccessRepositoryInterface $writeRepository
+     * @param ContactInterface $user
+     * @param ReadAccessGroupRepositoryInterface $accessGroupRepository
+     * @param DataStorageEngineInterface $dataStorageEngine
+     */
+    public function __construct(
+        private readonly ReadResourceAccessRepositoryInterface $readRepository,
+        private readonly WriteResourceAccessRepositoryInterface $writeRepository,
+        private readonly ContactInterface $user,
+        private readonly ReadAccessGroupRepositoryInterface $accessGroupRepository,
+        private readonly DataStorageEngineInterface $dataStorageEngine
+    ) {
+    }
+
+    /**
+     * @param int $ruleId
+     * @param PresenterInterface $presenter
+     */
+    public function __invoke(int $ruleId, PresenterInterface $presenter): void
+    {
+        /**
+         * Check if current user is authorized to perform the action.
+         * Only users linked to AUTHORIZED_ACL_GROUPS acl_group and having access in Read/Write rights on the page
+         * are authorized to add a Resource Access Rule.
+         */
+        if (! $this->isAuthorized()) {
+            $this->error(
+                "User doesn't have sufficient rights to delete a resource access rule",
+                [
+                    'user_id' => $this->user->getId(),
+                ]
+            );
+            $presenter->setResponseStatus(
+                new ForbiddenResponse(RuleException::notAllowed()->getMessage())
+            );
+
+            return;
+        }
+
+        try {
+            $this->info('Starting resource access rule deletion', ['rule_id' => $ruleId]);
+            if (! $this->readRepository->exists($ruleId)) {
+                $this->error('Resource access rule not found', ['rule_id' => $ruleId]);
+                $presenter->setResponseStatus(new NotFoundResponse('Resource access rule'));
+
+                return;
+            }
+
+            $this->deleteRule($ruleId);
+            $presenter->setResponseStatus(new NoContentResponse());
+        } catch (\Throwable $exception) {
+            $presenter->setResponseStatus(new ErrorResponse(RuleException::errorWhileDeleting($exception)));
+            $this->error($exception->getMessage(), ['trace' => $exception->getTraceAsString()]);
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    private function isAuthorized(): bool
```

Comment:
Can this method be globalized for all resource access usecases ? I think dashboard did something similar

Reply:
Nope. Each UseCase should have its own authorization method. It will avoid UseCases to inherit from other classes or UseCases...